internal/consumer/service_test.go

```diff
@@ -13,63 +14,92 @@ import (
+		dbMock        *database.Mock
+		ids           []int
+		config        *model.Configuration
+		expectedMocks func(t *testing.T, hnMock *hackernews.Mock, dbMock *database.Mock)
 	}{
 		"One Item": {
 			hnMock: &hackernews.Mock{},
+			dbMock: &database.Mock{},
 			ids:    []int{1},
-			consumerConfig: &model.ConsumerConfig{
-				BaseUrl:         "test.com",
-				CronSchedule:    "",
-				NumberOfWorkers: 2,
-			},
-			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock) {
+			config: &model.Configuration{
+				Consumer: model.ConsumerConfig{
+					BaseUrl:         "test.com",
+					CronSchedule:    "",
+					NumberOfWorkers: 2,
+				}},
+
+			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock, dbMock *database.Mock) {
 				hnMock.On("GetTopStories").Return([]int{1}, nil)
 				hnMock.On("GetItem", 1).Return(&hnModel.Item{ID: 1}, nil)
+				dbMock.On("SaveItem", &hnModel.Item{ID: 1}).Return(nil)
+
 			},
 		},
 		"Two Items": {
 			hnMock: &hackernews.Mock{},
+			dbMock: &database.Mock{},
 			ids:    []int{1},
-			consumerConfig: &model.ConsumerConfig{
-				BaseUrl:         "test.com",
-				CronSchedule:    "",
-				NumberOfWorkers: 2,
-			},
-			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock) {
+			config: &model.Configuration{
+				Consumer: model.ConsumerConfig{
+					BaseUrl:         "test.com",
+					CronSchedule:    "",
+					NumberOfWorkers: 2,
+				}},
+			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock, dbMock *database.Mock) {
 				hnMock.On("GetTopStories").Return([]int{1, 2}, nil)
 				hnMock.On("GetItem", 1).Return(&hnModel.Item{ID: 1}, nil)
 				hnMock.On("GetItem", 2).Return(&hnModel.Item{ID: 2}, nil)
+				dbMock.On("SaveItem", &hnModel.Item{ID: 1}).Return(nil).Once()
+				dbMock.On("SaveItem", &hnModel.Item{ID: 2}).Return(nil).Once()
 			},
 		},
-		"Error on item1": {
+		"Unable to get item from hackernews": {
 			hnMock: &hackernews.Mock{},
+			dbMock: &database.Mock{},
 			ids:    []int{1},
-			consumerConfig: &model.ConsumerConfig{
-				BaseUrl:         "test.com",
-				CronSchedule:    "",
-				NumberOfWorkers: 2,
-			},
-			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock) {
+			config: &model.Configuration{
+				Consumer: model.ConsumerConfig{
+					BaseUrl:         "test.com",
+					CronSchedule:    "",
+					NumberOfWorkers: 2,
+				}},
+			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock, dbMock *database.Mock) {
 				hnMock.On("GetTopStories").Return([]int{1, 2}, nil)
 				hnMock.On("GetItem", 1).Return(nil, errors.New("Failed to retrieve item"))
 				hnMock.On("GetItem", 2).Return(&hnModel.Item{ID: 2}, nil)
+				dbMock.On("SaveItem", &hnModel.Item{ID: 2}).Return(nil).Once()
+			},
+		},
+		"Unable save item": {
+			hnMock: &hackernews.Mock{},
+			dbMock: &database.Mock{},
+			ids:    []int{1},
+			config: &model.Configuration{
+				Consumer: model.ConsumerConfig{
+					BaseUrl:         "test.com",
+					CronSchedule:    "",
+					NumberOfWorkers: 2,
+				}},
```

Comment:
NIT: these could be indented

Reply:
No sure what you mean by that?

all the format is done using go fmt

Reply:
if you run `gofumpt -w -extra . ` you will see the below...

```
		"One Item": {
			hnMock: &hackernews.Mock{},
			dbMock: &database.Mock{},
			ids:    []int{1},
			config: &model.Configuration{
				Consumer: model.ConsumerConfig{
					BaseUrl:         "test.com",
					CronSchedule:    "",
					NumberOfWorkers: 2,
				},
			},

			expectedMocks: func(t *testing.T, hnMock *hackernews.Mock, dbMock *database.Mock) {
				hnMock.On("GetTopStories").Return([]int{1}, nil)
				hnMock.On("GetItem", 1).Return(&hnModel.Item{ID: 1}, nil)
				dbMock.On("SaveItem", &hnModel.Item{ID: 1}).Return(nil)
			},
		},
```